OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -643,8 +655,8 @@ static class DMOpcodeHandlers {
                     result = second;
                 }
             } else if (second.Value != null) {
-                if (first.Type == DreamValue.DreamValueType.Float && second.Type == DreamValue.DreamValueType.Float) {
-                    result = new DreamValue(first.GetValueAsInteger() | second.GetValueAsInteger());
+                if (first.TryGetValueAsInteger(out var firstInt) && second.TryGetValueAsInteger(out var secondInt)) {
```

Comment:
I'm assuming this is equivalent behavior but I can re-add the typechecks if not.

Reply:
It is equivalent behavior.

Reply:
It is equivalent behavior.